README.md
# Projet-Mares_BDD-Environnementale
Analyse et exploitation de la BDD Environnementale du Projet Mares 2017


Travail_MrMargerie/history.txt
load("Z:/Analyses Statistiques/Projet mares/Workspace 16 04.RData")
ls
ls()
View(Amfreville)
View(bdd)
summar(bddw32)
summary(bddw32)
dudi.acm(bddw32)
library(ade4)
dudi.acm(bddw32)
dim(bddw32)
View(bddw32)
colnames(bddw32)
acm32 <- dudi.acm(bddw32)
scatter(acm32)
acm32$tab
dim(acm32$tab)
s.label(acm32$li)
plot(hclust(dist(acm32$li)))
dist(acm32$li)
dim(acm32$tab)
head(acm32$li)
head(dist(acm32$li))
cutree(hclust(dist(acm32$li)), k=4)
dim(cutree(hclust(dist(acm32$li)), k=4))
length(cutree(hclust(dist(acm32$li)), k=4))
cutree(hclust(dist(acm32$li)), k=4) -> cut4
s.class(acm32$li, cut4)
s.class(acm32$li, factor(cut4))
table(bddw32[,1])
table(bddw32[,1], cut4)
table(bddw32[,2], cut4)
chisq.test(bddw32[,1], cut4)
chisq.test(table(bddw32[,2], cut4))
chisq.test(bddw32[,1], cut4)
plot(hclust(dist(acm32$li)))
pam(acm32$li, k=2)
?pam
??pam
libary(clusters)
library(clusters)
library(cluster)
pam(acm32$li, k=2)
plot(pam(acm32$li, k=2))
plot(pam(acm32$li, k=3))
plot(pam(acm32$li, k=4))
plot(pam(acm32$li, k=5))
plot(pam(acm32$li, k=20))
pam3 <- pam(acm32$li, k=3)
pam3
s.class(acm32$li, pam3$clustering)
s.class(acm32$li, factor(pam3$clustering))
pam3$medoids
pam3$medois[pam3$medoids==3]
pam3$medoids[pam3$medoids==3]
pam3$medoids[pam3$medoids=="3"]
acm32$li[pam3$clustering==1]
acm32$li[pam3$clustering==1,]
acm32$li[pam3$clustering==1,] -> p31
pam(p31, k=2)
plot(pam(p31, k=2))
plot(pam(p31, k=3))
plot(pam(p31, k=4))
pam(p31, k=4) -> pam314
s.class(acm32$li, factor(pam3$clustering))
s.class(p31, factor(pam314$clustering))
s.class(p31, factor(pam314$clustering), add.plot=TRUE)
s.class(acm32$li, factor(pam3$clustering))
s.class(p31, factor(pam314$clustering), add.plot=TRUE)
s.class(acm32$li, factor(pam3$clustering))
s.class(p31, factor(pam314$clustering), add.plot=TRUE, col="red")
s.class(acm32$li, factor(pam3$clustering))
s.class(p31, factor(pam314$clustering), add.plot=TRUE, col=c(rep("red", 4)))
savehistory("Z:/Analyses Statistiques/Projet mares/history.txt")


StatistiqueExploratoire.R
library(tidyverse)

dat <- read.table("bdd_vf.txt", sep="\t", head=T, na.strings = "NC", row.names=1, dec=",")

id.numbers <- rownames(dat)
dat <- as.tibble(dat)

dat2 <- dat %>%
  add_column(ID = id.numbers) %>%
  select(ID,
         Type = type,
         Forme = forme,
         Longueur,
         Largeur = largeur,
         Contexte = contexte_1,
         Topo = topo,
         Liaison_Hydro = liaison_reseau,
         Alim_Spe = alim_specifique,
         Regime = regime,
         Turbidite = turbidite,
         Profondeur = profondeur,
         Berges = berges,
         Pietinement = pietinement,
         Boisement = boisement,
         Ombrage = ombrage,
         Helophytes = helophytes,
         Hydrophytes_Enr = hydrophytes_enracines,
         Hydrophytes_NEnr = hydrophytes_non_enracines,
         Algues = algues,
         Fond_Ex = fond_exonde,
         Eau_Libre = eau_libre,
         Evolution = evolution,
         Poissons = poisson,
         Amphibiens = amphibien,
         Odonates = odonate,
         Invasives = presence_flore_exotique,
         Fond = nature_fond,
         Commentaire = commentaire) %>%
  filter(Commentaire != "mare_comblee" | is.na(Commentaire)) %>%
  filter(Commentaire != "mare_seche" | is.na(Commentaire)) %>%
  filter(Commentaire != "mare_seche_ou_pas_trouvee" | is.na(Commentaire)) %>%
  filter(!is.na(Forme)) %>%
  filter(!is.na(Longueur))

ggplot(dat2, aes(x = Contexte)) + geom_bar(aes(y = Fond_Ex), stat = "identity")


# Test pour représenter le recouvrement en végétation selon le milieu

rec <- dat2 %>%
  select(Contexte,
         Evolution,
         Profondeur,
         Topo,
         Helophytes,
         Hydrophytes_Enr,
         Hydrophytes_NEnr,
         Algues,
         Fond_Ex,
         Eau_Libre)

rec2 <- rec %>%
  gather(Helophytes,
         Hydrophytes_Enr,
         Hydrophytes_NEnr,
         Algues,
         Fond_Ex,
         Eau_Libre,
         key = "Vegetation",
         value = "Recouvrement")
rec2 <- rec2 %>%
  mutate(
    Vegetation = fct_relevel(as.factor(Vegetation), c("Helophytes", "Hydrophytes_Enr", "Hydrophytes_NEnr", 
                                                      "Algues", "Eau_Libre", "Fond_Ex"))
  )

Recouvrement <- ggplot(rec2) + 
  geom_bar(aes(x = Evolution, y = Recouvrement, fill = Vegetation), stat = "identity", position = "fill") +
  ggtitle("Recouvrement des mares selon le stade d'évolution et le type de Végétation") +
  theme(plot.title = element_text(hjust = .5, size = 24))
Recouvrement

Prof <- ggplot(rec2, aes(x = Profondeur, y = Recouvrement, col = Vegetation)) +
  geom_point() +
  geom_smooth(se = F)
Prof

















Mise_en_forme.R
library(tidyverse)

dat <- read.table("bdd_vf.txt", sep="\t", head=T, na.strings = "NC", row.names=1, dec=",")

load("Travail_MrMargerie/Workspace 16 04.RData")
# On s'intéresse ici à bddw32, qui servira de référence


# On vire les variables non utilisées par la promo précédente
# Seul problème : eux ils ont des données qu'on n'a pas ...

id.numbers <- rownames(dat)
dat <- as.tibble(dat)

dat2 <- dat %>%
  add_column(ID = id.numbers) %>%
  select(ID = ID,
         TYPE = type,
         FORME = forme,
         TAIL_LONG = Longueur,
         TAIL_LARG = largeur,
         CONTEXTE = contexte_1,
         TOPO = topo,
         LIAI_HYDRO = liaison_reseau,
         ORIG_EAU = alim_specifique,
         REG_HYDRO = regime,
         EAU = turbidite,
         #DISTANCE = ?
         PROF_TO = profondeur,
         #PROF_MAX = ?
         PENTEDOUC = berges,
         PIETINEMEN = pietinement,
         BOISE = boisement,
         OMBRAGE = ombrage,
         HELOPHYTE = helophytes,
         HYDROPHYTR = hydrophytes_enracines,
         HYDRO2 = hydrophytes_non_enracines,
         #HYDRO3 = ?
         #HYDRO4 = ?
         ALGUES = algues,
         EAUHYDRO = fond_exonde,
         EAU_LIBRE = eau_libre,
         EVOL_MARE = evolution,
         POISSONS = poisson,
         AMPHIBIENS = amphibien,
         ODONATES = odonate,
         INVASIVES = presence_flore_exotique,
         #POT_ECO = ?
         Fond = nature_fond)


# On fait alors en sorte d'avoir les mêmes modalités de variables partout :

dat3 <- dat2 %>%
  mutate(
    TYPE = fct_recode(TYPE,
                      "BASSIN" = "bassin",
                      "MARE" = "mare"),
    FORME = fct_recode(FORME,
                      "carre/rectangle" = "carre_rectangle",
                      "carre/rectangle" = "triangle",
                      "complexe" = "complexe",
                      "complexe" = "patatoide",
                      "ronde/ovale" = "ronde_ovale"),
    CONTEXTE = fct_recode(CONTEXTE,
                          "annexes_routieres" = "annexe_routiere_ferroviere",
                          "bois_feuillus" = "bois_de_feuillus",
                          "bois_feuillus" = "fourres_bosquets",
                          "culture" = "culture",
                          "parmi_habitations/milieux_urbains" = "jardin_parc_cour_de_ferme",
                          "prairie_mesophile" = "prairie_humide",
                          "prairie_mesophile" = "lande_humide",
                          "tourbière" = "bas_marais_tourbiere_alcaline",
                          "je_ne_sais_pas" = "indetermine"),
    TOPO = fct_recode(TOPO,
                      "fond_vallee" = "fond_de_vallee"),
    LIAI_HYDRO = fct_recode(LIAI_HYDRO,
                            "ecoulement_a_sec_en_sortie" = ,
                            "ecoulement_actif_en_sortie" = ,
                            "jamais_decoulement_en_sortie" = ),
  )




